armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -387,6 +387,119 @@ def getNumberDensityWithTrace(component, nucName):
     return nvt, nv
 
 
+class CylindricalComponentsAverageBlockCollection(BlockCollection):
+    """
+    Creates a representative block for the purpose of cross section generation with a one-dimensional
+    cylindrical model.
+
+    Notes
+    -----
+    When generating the representative block within this collection, the geometry is checked
+    against all other blocks to ensure that the number of components are consistent. This implementation
+    is intended to be opinionated, so if a user attempts to put blocks that have geometric differences
+    then this will fail.
+
+    This selects a representative block based on the collection of candidates based on the
+    median block average temperatures as an assumption.
+    """
+
+    def _getNewBlock(self):
+        newBlock = copy.deepcopy(self._selectCandidateBlock())
+        newBlock.name = "1D_CYL_AVG_" + newBlock.getMicroSuffix()
+        return newBlock
+
+    def _selectCandidateBlock(self):
+        """Selects the candidate block with the median block-average temperature."""
+        info = []
+        for b in self.getCandidateBlocks():
+            info.append((b.getAverageTempInC() * self.getWeight(b), b.getName(), b))
```

Comment:
Why is the temperature multiplied by the weight here? I wouldn't expect the weight to play any role in finding the median.

Reply:
Yeah I just copied and pasted from another class. The weight is just 1.0 for all blocks I believe so you're correct